src/ssl/ErrorDetailManager.cc

```diff
@@ -24,6 +26,24 @@ void Ssl::errorDetailClean()
     Ssl::ErrorDetailsManager::Shutdown();
 }
 
+/// ErrorDetailEntry constructor helper that extracts a quoted HTTP field value
+static SBuf
+SlowlyParseQuotedField(const char * const description, const HttpHeader &parser, const char * const fieldName)
+{
+    String fieldValue;
+    parser.hasNamed(fieldName, strlen(fieldName), &fieldValue);
```

Comment:
Calling hasNamed() without checking its result does not make much sense. Before we decide how to write this code, could you please explain what was the motivation behind commit 088a048c92fd49daa1606b686edfd613a39c0d09? Commit message does not answer that question AFAICT.

I am aware of the following two statements, but they do not help me answer that "why commit https://github.com/squid-cache/squid/commit/088a048c92fd49daa1606b686edfd613a39c0d09" question:

>> Alex at https://github.com/squid-cache/squid/pull/1451#discussion_r1337658245: This check is missing in the official code. Without this check httpHeaderParseQuotedString() crashes in the unlikely event an error-details.txt entry is missing a required "detail" or "descr" field. I have disclosed this fix in the updated PR description.

>> httpHeaderParseQuotedString() also has parsing bugs. I do not think we should touch that old function in this PR. I also do not think we should adjust our SlowlyParseQuotedString() wrapper to compensate for some of those httpHeaderParseQuotedString() problems. The wrapper should remain a wrapper.

> Francesco at https://github.com/squid-cache/squid/pull/1451#discussion_r1347716925: Okay; I notice that Http::SlowlyParseQuotedString has some checking and uses exceptions for the fail path, so it seems compatible with the intent here. At this point, this wrapper is useless at all, I can just call Http::SlowlyParseQuotedString, as this function only has two callsites.

Reply:
hasNamed() is misnamed; the third argument is an OUT parameter, which returns into the provided string, maybe this is the source of the confusion. Can we do without checking the return value? Yes, because if the header is not found, an empty string will be returned, and Http::SlowlyParseQuotedString will call httpHeaderParseQuotedString which will fail because the first character is not a ```"``` .

I had interpreted your request in this way; but yes, if you ask me, one more free check would help

Reply:
I misinterpreted your comment as a suggestion to remove the check. Accepted your suggestion reinstating it, I'm good with the code as is. I don't think there's any more pending requests for improvements, are there?

Reply:
> I don't think there's any more pending requests for improvements, are there?

I do not see any. Please feel free to clear for merge when you are comfortable with PR code.